### DISCUSSION TITLE
[Just asking - low priority] Add download function for filesystem

### DISCUSSION THREAD

**Discussion by hungcrush** [pocketbase/pocketbase#1478] [Q&A]
Hi, I'm working with Media Library, and I want to add feature upload by URL. 
If you have time, could you give me your opinion about bellow code 😬 

```go
func (s *System) DownloadFileByUrl(url string) (string, error) {
	// fetch the file
	resp, err := http.Get(url)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	// get the file name from the url
	getFileNameFromUrl := func(url string) string {
		parts := strings.Split(url, "/")
		return parts[len(parts)-1]
	}

	// create a new file
	fileName := "media/" + getFileNameFromUrl(url)

	w, _ := s.bucket.NewWriter(s.ctx, fileName, nil)

	// write the body to file
	_, err = io.Copy(w, resp.Body)

	if err != nil {
		return "", err
	}

	// check for close errors to ensure that the file was really saved
	return fileName, w.Close()
}
```

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the code sample.

If `System` in the above is `filesystem.System` I guess you have a custom fork, right?

It depends what you are trying to do, but the recommended way to extend PocketBase is to [use it as framework](https://pocketbase.io/docs/use-as-framework/) (aka. embed it in your Go app as any other package).

If your use case requires modifying the source, then I guess the above is ok if you are trying to upload to S3 from a url (_although you may want to add some validations for the mimetype, size, etc._)

**Reply by hungcrush** [reply]
Thanks for your time. I got it. 

I'm learning by reading and modify your source. Just hope to be as good as you 😬

**Reply by hungcrush** [reply]
Show off to you I'm doing Media Library 😬 and I intend to add more Collection Field: `media`
<img width="1657" alt="CleanShot 2023-01-03 at 14 31 27@2x" src="https://user-images.githubusercontent.com/8098916/210315433-88a486d7-ec3a-49f7-af6a-7f4c5bfb492c.png">
